### ISSUE TITLE
Huge build times

### ISSUE THREAD

**Issue by dariooddenino** [alexmingoia/pux-starter-app#14]
Hi,
I've just cloned this repo and started it following the readme.

My problem is that each build takes ~30/40 seconds, even on hot reloads. Every time I get something between 150 and 750 warnings.

Basically it stays on the `webpack building...` message for those 30/40 seconds and then dumps all the warnings and the `bundle is now VALID` message.

I'm on Ubuntu 15.10, psci is version 0.8.3.0 (if it matters)

Thanks!

**Comment by alexmingoia** [comment]
You should only be getting warnings on the first build. Subsequent builds in watch mode shouldn't trigger a huge amount of warnings. The warnings are from the purescript compiler, and are because most modules haven't been updated to using specified imports which started generating warnings in purescript 0.8.

The build times are a separate issue. There's two problems:
- The first is that builds are triggered twice due to a bug in how `purs-loader` and `purescript-webpack-plugin` works. I have written my own loader which solves this and I should be publishing that very soon. 
- A full purescript recompile is being done on every webpack build. This can be sped up by using new per-file rebuild features in purescript 0.8. My loader that I'm releasing soon implements this as well. 

So TLDR hang tight and I should have both those issues resolved.

**Comment by dariooddenino** [comment]
Hi, thanks for the answer!

I'm confused now as I had people on slack telling me they had around 2/3 seconds build time on hot reload! 

Yeah, I noticed the the build is triggered twice. The first one takes most of the time (pratically all of it) and the second one is < 1 sec.

Thanks for your hard work!

**Comment by shybyte** [comment]
@alexmingoia I'm looking forward to the new loader with per-file-build-speedup. Currently I have a re-build time of ~11 seconds on the unmodified starter-app, which neutralizes the advantages of hot reloading a bit. Thanks for your great work!

**Comment by alexmingoia** [comment]
An update on the new loader: I've made a PR to the existing `purs-loader` module: ethul/purs-loader/pull/47

If that's not accepted I'll release it as a separate module and update the starter-app accordingly.

**Comment by alexmingoia** [comment]
The new release is using `purs-loader@1.0.0-rc.1`, which should be much quicker, solves the double-compilation issue, and has optional experimental support for instant-rebuilds using psc-ide (https://github.com/ethul/purs-loader#instant-rebuilds-experimental)
